src/commands/createWebApp/setDefaultRgAndPlanName.ts

```diff
@@ -12,28 +12,32 @@ import { createAzureClient, DialogResponses, IActionContext } from "vscode-azure
 import { ext } from "../../extensionVariables";
 import { nonNullProp } from "../../utils/nonNull";
 import { getWorkspaceSetting, updateGlobalSetting } from "../../vsCodeConfig/settings";
+import { AzConfig, AzConfigProperties, readAzConfig } from "./readAzConfig";
 
 const maxNumberOfSites: number = 3;
 
 export async function setDefaultRgAndPlanName(wizardContext: IAppServiceWizardContext, siteNameStep: SiteNameStep): Promise<void> {
+    const config: AzConfig = await readAzConfig(AzConfigProperties.group, AzConfigProperties.plan);
+
     // this should always be set when in the basic creation scenario
     const location: Location = nonNullProp(wizardContext, 'location');
-    const defaultName: string = `appsvc_${wizardContext.newSiteOS}_${location.name}`;
+    const defaultGroupName: string = config.group ? config.group : `appsvc_${wizardContext.newSiteOS}_${location.name}`;
+    const defaultPlanName: string = config.plan ? config.plan : defaultGroupName;
 
-    const asp: AppServicePlan | null = await getAppServicePlan(wizardContext, defaultName, defaultName);
+    const asp: AppServicePlan | null = await getAppServicePlan(wizardContext, defaultGroupName, defaultPlanName);
```

Comment:
Have you tried this test case?
- Create a plan with your default azconfig plan name in a random rg
- Try to create a new app with your azconfig plan name, but don't specific an azconfig rgName

Reply:
This case passes. It allows the same plan name in different resource groups. Setting the azconfig plan but not the rg defaults to use the `appsvc_<os>_<location>` rg name.